src/test/metrics_flow_test.cpp

```diff
@@ -0,0 +1,605 @@
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_streams{name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(4)));
+    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_streams{name=\""} + dagName + std::string{"\",version=\"1\"} "})));
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_infer_req_queue_size{name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(2)));
+    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_infer_req_queue_size{name=\""} + dagName + std::string{"\",version=\"1\"} "})));
+}
+
+TEST_F(MetricFlowTest, GrpcGetModelMetadata) {
+    PredictionServiceImpl impl(server);
+    tensorflow::serving::GetModelMetadataRequest request;
+    tensorflow::serving::GetModelMetadataResponse response;
+
+    for (int i = 0; i < numberOfSuccessRequests; i++) {
+        request.Clear();
+        response.Clear();
+        request.mutable_model_spec()->mutable_name()->assign(modelName);
+        request.add_metadata_field("signature_def");
+        ASSERT_EQ(impl.GetModelMetadata(nullptr, &request, &response).error_code(), grpc::StatusCode::OK);
+    }
+
+    for (int i = 0; i < numberOfSuccessRequests; i++) {
+        request.Clear();
+        response.Clear();
+        request.mutable_model_spec()->mutable_name()->assign(dagName);
+        request.add_metadata_field("signature_def");
+        ASSERT_EQ(impl.GetModelMetadata(nullptr, &request, &response).error_code(), grpc::StatusCode::OK);
+    }
+
+    checkRequestsCounter(server.collect(), "ovms_requests_success", modelName, 1, "gRPC", "GetModelMetadata", "TensorFlowServing", numberOfSuccessRequests);  // ran by real request
+    checkRequestsCounter(server.collect(), "ovms_requests_success", dagName, 1, "gRPC", "GetModelMetadata", "TensorFlowServing", numberOfSuccessRequests);    // ran by real request
+}
+
+TEST_F(MetricFlowTest, GrpcGetModelStatus) {
+    ModelServiceImpl impl(server);
+    tensorflow::serving::GetModelStatusRequest request;
+    tensorflow::serving::GetModelStatusResponse response;
+
+    for (int i = 0; i < numberOfSuccessRequests; i++) {
+        request.Clear();
+        response.Clear();
+        request.mutable_model_spec()->mutable_name()->assign(modelName);
+        ASSERT_EQ(impl.GetModelStatus(nullptr, &request, &response).error_code(), grpc::StatusCode::OK);
+    }
+
+    for (int i = 0; i < numberOfSuccessRequests; i++) {
+        request.Clear();
+        response.Clear();
+        request.mutable_model_spec()->mutable_name()->assign(dagName);
+        ASSERT_EQ(impl.GetModelStatus(nullptr, &request, &response).error_code(), grpc::StatusCode::OK);
+    }
+
+    checkRequestsCounter(server.collect(), "ovms_requests_success", modelName, 1, "gRPC", "GetModelStatus", "TensorFlowServing", numberOfSuccessRequests);  // ran by real request
```

Comment:
It's probably little late to do it before RC but wouldn't it be better to have consts labels declared eg. in metriclabels.hpp
const std::string REQUEST_SUCCESS = "ovms_requests_success";
const std::string TFS_MODEL_STATUS_LABEL = "GetModelStatus";
etc.

To avoid possible misspelings? Possibly to be done in cleanup later on not in this MR.

Reply:
Will add task for that